__init__.py

```diff
@@ -1,68 +1,88 @@
+    def __init__(self, length: int, characters: str) -> None:
+        self.length: int = length
+        self.characters: str = characters
 
-class complexpass(simplepass):
-    def __init__(self, length, string_method, numbers=True, special_chars=False):
-        self.length = length
-        self.string_method = string_method
-        self.numbers = numbers
-        self.special_chars = special_chars
-
-    def generate(self, iterations):
-        '''
-        Generates a string for the var characters that is used to make 
-        a password depending on length, upper of lowercase, digits, special
-        characters and iterations. Uses 2 for loops to achieve this.
-        '''
-        
-        characters = ''
+    def generate(self, iterations: int = 1) -> Union[str, List[str]]:
+        """returns one or multiple passwords,
+        depending on length, given characters and iterations.
 
-        methods = {
-            'upper': string.ascii_uppercase,
-            'lower': string.ascii_lowercase,
-            'both': string.ascii_letters
-        }
+        Args:
+            iterations (int): how many passwords to return
+
+        Returns:
+            Union[str, List[str]]: one or multiple passwords
+        """
+        if iterations == 1:
+            return self._gen_pw()
+        else:
+            return [self._gen_pw() for _ in range(iterations)]
+
+    def _gen_pw(self) -> str:
+        """returns a password depending on lenght and given caracters
+
+        Returns:
+            str: a randomly generated password
+        """
+        output_password: str = ""
+        for _ in range(self.length):
+            output_password += secrets.choice(self.characters)
+        return output_password
 
-        characters += methods[self.string_method]
 
-        if self.numbers:
+class complexpass(simplepass):
+    def __init__(
+        self,
+        length: int,
+        string_method: str = "both",
+        numbers: bool = True,
+        special_chars: bool = False,
+    ):
+        methods = {
+            "upper": string.ascii_uppercase,
+            "lower": string.ascii_lowercase,
+            "both": string.ascii_letters,
+        }
+        characters: str = methods[string_method]
+        if numbers:
             characters += string.digits
-        if self.special_chars:
+        if special_chars:
             characters += string.punctuation
-            
-        # Generates the password
+        super().__init__(length=length, characters=characters)
 
-        for p in range(iterations):
-            output_password = ''
-            for c in range(self.length):
-                output_password += secrets.choice(characters)
-            print(output_password)
 
 # Tests
-
-if __name__ == '__main__':
+if __name__ == "__main__":
     # Generate a complex password
-    
-    var1 = complexpass(20, 'both', True, True)
-    var1.generate(3)
-    
+
+    cp = complexpass(20, "both", True, True)
+    print(cp.generate())
```

Comment:
cp.generate() uses the method of the Simplepass class. This is because, Complexpass is just an extension of Simplepass. One might say, that Complexpass IS a Simplepass. Thus it has its attributes (also methods for that matter)